daemon/cmd/endpoint.go

```diff
@@ -174,7 +174,15 @@ func (d *Daemon) fetchK8sLabelsAndAnnotations(nsName, podName string) (*slim_cor
 	return p, containerPorts, identityLabels, infoLabels, annotations, nil
```

Comment:
In this file, instead of the proposed changes I suggest to do the following (on all error conditions):
```
		ep.SetState(endpoint.StateInvalidEndpoint, "Invalid endpoint")
```

and in `pkg/endpoint/endpoint.go`
```patch
diff --git a/pkg/endpoint/endpoint.go b/pkg/endpoint/endpoint.go
index 8bd438963..bd504c97a 100644
--- a/pkg/endpoint/endpoint.go
+++ b/pkg/endpoint/endpoint.go
@@ -96,6 +96,8 @@ const (
        // StateRestoring is used to set the endpoint is being restored.
        StateRestoring = string(models.EndpointStateRestoring)
 
+       // TODO Needs an new state in the api/v1/models`
+       StateInvalidEndpoint = string(models.StateInvalidEndpoint)
+
        // IpvlanMapName specifies the tail call map for EP on egress used with ipvlan.
        IpvlanMapName = "cilium_lxc_ipve_"
 )
@@ -1319,7 +1321,7 @@ func (e *Endpoint) setState(toState, reason string) bool {
                }
        case StateWaitingForIdentity:
                switch toState {
-               case StateReady, StateDisconnecting:
+               case StateReady, StateDisconnecting, StateInvalidEndpoint:
                        goto OKState
                }
        case StateReady:
@@ -1389,7 +1391,7 @@ OKState:
 
        // Since StateDisconnected is the final state, after which the
        // endpoint is gone, we should not increment metrics for this state.
-       if toState != "" && toState != StateDisconnected {
+       if toState != "" && (toState != StateDisconnected || toState != StateInvalidEndpoint) {
                metrics.EndpointStateCount.
                        WithLabelValues(toState).Inc()
        }
@@ -1403,7 +1405,7 @@ func (e *Endpoint) BuilderSetStateLocked(toState, reason string) bool {
        // Validate the state transition.
        fromState := e.state
        switch fromState { // From state
-       case StateWaitingForIdentity, StateReady, StateDisconnecting, StateDisconnected:
+       case StateWaitingForIdentity, StateReady, StateDisconnecting, StateDisconnected, StateInvalidEndpoint:
                // No valid transitions for the builder
        case StateWaitingToRegenerate, StateRestoring:
                switch toState {
@@ -1456,7 +1458,7 @@ OKState:
 
        // Since StateDisconnected is the final state, after which the
        // endpoint is gone, we should not increment metrics for this state.
-       if toState != "" && toState != StateDisconnected {
+       if toState != "" && (toState != StateDisconnected || toState != StateInvalidEndpoint) {
                metrics.EndpointStateCount.
                        WithLabelValues(toState).Inc()
        }
```